crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -132,6 +132,10 @@ pub struct Room {
     /// to disk but held in memory.
     #[cfg(all(feature = "e2e-encryption", feature = "experimental-sliding-sync"))]
     pub latest_encrypted_events: Arc<SyncRwLock<RingBuffer<Raw<AnySyncTimelineEvent>>>>,
+
+    /// The list of pinned event ids in this room in an observable format so
+    /// they can be subscribed to.
+    pub(crate) pinned_event_ids: SharedObservable<Vec<OwnedEventId>>,
```

Comment:
So the reason why we usually don't put things in this struct is that they're not persisted after a reload; if you reload a room, you'd have to wait for the next sync to arrive to give you the list of events, and if you happen to be offline, that means that now you don't have access to pinned events.

From here, we have a few possibilities:

- include the `Vec<OwnedEventId>` in `RoomInfo`,
- include the pinned event in the `BaseRoomInfo`, and have a getter to read the pinned event ids from there
- have a new (de)serializable field in `Room` that is filled at startup, and updated as the events come. You've done half of the work here by having it be updated as the events come, and we'd need some way to serialize/deserialize it too later.

My preference would likely go to the first or second option (second option might be faster to ser/de), to benefit from persistence. Can you also add a test that would've failed without this change, please?

Reply:
Ok, I tried fixing this in https://github.com/matrix-org/matrix-rust-sdk/pull/3747/commits/4aa51cada2be754cddbd6f88f1b2fe68c5a5ef7c using the 2nd strategy 🤞 .